wrappers/pkcs11/pkcs11.go

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package pkcs11
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+	"sync/atomic"
+
+	uuid "github.com/hashicorp/go-uuid"
+	pkcs11 "github.com/miekg/pkcs11"
+	wrapping "github.com/openbao/go-kms-wrapping/v2"
+)
+
+// These constants contain the accepted env vars; the Vault one is for backwards compat
+const (
+	EnvPkcs11WrapperKeyId   = "PKCS11_WRAPPER_KEY_ID"
+	EnvVaultPkcs11SealKeyId = "VAULT_PKCS11_SEAL_KEY_ID"
```

Comment:
Do you want to name this `BAO_` and use [the helper](https://pkg.go.dev/github.com/openbao/openbao/api/v2#ReadBaoVariable) to read the value?

Also, I think [per docs](https://developer.hashicorp.com/vault/docs/configuration/seal/pkcs11#pkcs11-environment-variables) this should be called `VAULT_HSM_KEY_ID` and other environment variables shouldn't be `PKCS11_...` but instead `BAO_HSM_...`  (which will then be `VAULT_` with the helper). :-)